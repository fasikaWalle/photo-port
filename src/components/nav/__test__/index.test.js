import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "..";

afterEach(cleanup);
describe("Nav component", () => {
  // baseline test
  it("renders", () => {
    render(<Nav />);
  });
  it("matches snapshot", () => {
    const { asFragment } = render(<Nav />);

    expect(asFragment()).toMatchSnapshot();
  });
});

describe("emoji is visible", () => {
  it("insert emoji into the h2", () => {
    const { getByLabelText } = render(<Nav />);
    expect(getByLabelText("camera")).toHaveTextContent("ðŸ“¸");
  });
});
describe("link is visible", () => {
  it("insert text into the link", () => {
    const { getByTestId } = render(<Nav />);
    expect(getByTestId("link")).toHaveTextContent("Oh Snap!");
    expect(getByTestId("about")).toHaveTextContent("About me");
  });
});


