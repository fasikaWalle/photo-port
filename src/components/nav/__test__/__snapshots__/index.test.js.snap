// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        <span
          aria-label="camera"
          role="img"
        >
           ðŸ“¸
        </span>
         Oh Snap!
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About me
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <h1>
          <span>
            commercial
          </span>
        </h1>
        <h1>
          <span>
            portraits
          </span>
        </h1>
        <h1>
          <span>
            food
          </span>
        </h1>
        <h1>
          <span>
            landscape
          </span>
        </h1>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
